imap/message.c

```diff
@@ -4956,7 +4969,8 @@ static int body_foreach_section(struct body *body, struct message *message,
             msg.len = body->header_size;
             msg.offset = 0;
             msg.encode = 0;
-            message_parse_headers(&msg, tmpbody, "text/plain", &boundaries, NULL);
+            message_parse_headers(&msg, tmpbody, "text/plain", &boundaries,
```

Comment:
Kinda surprised we don't care about errors in parsing here

Reply:
Here, we parse the headers for on an already parsed body-part here, e.g. these headers already were validated otherwise that body part wouldn't exist. We didn't check for errors previously here, because message_parse_headers didn't check for any.

It might be good to check for message_parse_headers errors here, too, but in case of errors it's a very different situation we are in: it means that an already accepted email on disk could now not be read anymore. We would need to treat this differently and I'd prefer to keep the damage radius of this change small.